tensorflow/python/keras/backend.py

```diff
@@ -4541,7 +4541,13 @@ def relu(x, alpha=0., max_value=None, threshold=0):
   Returns:
       A tensor.
   """
-
+  if isinstance(x, (ops.Tensor,
+                    variables_module.Variable,
+                    sparse_tensor.SparseTensor,
+                    np.ndarray)):
+    dtype = x.dtype
```

Comment:
Just curious why we need to do the type check here, shouldn't it always use the input dtype?

Reply:
I thought so too, but in some of the tests for this, they pass in a list/tuple which doesn't have a dtype defined, so in those cases we default to float32.

Reply:
I see. Then should we just check if 'x' has dtype attribute? I would expect the list of types here to grow when we have more types. Have to main the list here will be a maintenance headache.

Reply:
@qlzh727 changed it to check if the dtype attribute is available instead.

Reply:
Thanks. How about:

```
dtype = getattr(x, 'dtype', floatx())
```

Reply:
Please also add a comment for those x input which doesn't have dtype, eg raw list and tuple.

Reply:
updated @qlzh727